src/Analyzers/Core/Analyzers/FileHeaders/FileHeader.cs

```diff
@@ -104,5 +109,20 @@ internal Location GetLocation(SyntaxTree syntaxTree)
 
             return Location.Create(syntaxTree, new TextSpan(_fileHeaderStart, _commentPrefixLength));
         }
+
+        /// <summary>
+        /// Gets the location representing the file header.
+        /// </summary>
+        /// <param name="syntaxTree">The syntax tree to use for generating the location.</param>
+        /// <returns>The location representing the file header.</returns>
+        internal Location GetHeaderLocation(SyntaxTree syntaxTree)
+        {
+            if (IsMissing)
+            {
+                return Location.Create(syntaxTree, new TextSpan(_fileHeaderStart, 0));
+            }
+
+            return Location.Create(syntaxTree, TextSpan.FromBounds(_fileHeaderStart, _fileHeaderEnd));
```

Comment:
why does this need to return a Location, instead of just a TextSpan?

Reply:
Done in 099dcdd4